src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -46,11 +46,18 @@ internal bool IsUnprocessedBufferEmpty()
 
         internal void AppendExtraBuffer(ReadOnlySpan<byte> buffer)
         {
-            // Ensure a reasonable upper bound applies to the stackalloc
             Debug.Assert(buffer.Length <= 1024);
 
-            // Then convert the bytes to chars
-            Span<char> chars = stackalloc char[_encoding.GetMaxCharCount(buffer.Length)];
+            // The maximum buffer size that can be passed is 1024 in length.
+            // For UTF-8, GetMaxCharCount(1024) is 1025, so that much is expected
+            // on the stack since we want to avoid allocating the array.
+            // For custom encodings where the behavior of GetMaxCharCount is
+            // not known beforehand, this may fall back to an array.
+            const int MaxStackAllocation = 1025;
```

Comment:
1025 chars on the stack is steep in my opinion, but it is preserving the existing behavior.

This preserves the behavior of where previously we were potentially allocation that much on the stack anyway.

If we want to adjust the behavior, we can:

1. Make this value smaller, potentially allocating an array more.
2. We could rent a char array from a pool.
3. Make the buffer smaller.

Reply:
@stephentoub in case you have a thought on how to proceed there.

Reply:
We should use a smaller bound, like 512.  Can we use ArrayPool for things larger than that?

Reply:
That said, I'd need to re-review the code paths that get here, but it'd be really rare to have anything near to that large, right? This is only used for non-redirected inputs, so someone typing at the console, and this extra buffer functionality is only used as part of cursor-related operations, yes?

Reply:
Okay. The only place where this is passed a ROS bigger than "one" is here:

https://github.com/dotnet/runtime/blob/0e18cfda13a8136103c83442e1a649cdecc237b0/src/libraries/System.Console/src/System/IO/StdInReader.cs#L463

The only other call is in `ReadKey`, so I would make the same conclusion that the pool can be skipped.